roles/ci_dcn_site/tasks/ceph.yml

```diff
@@ -0,0 +1,91 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Fetch network facts of ceph bootstrap node
+  ansible.builtin.setup:
+    gather_subset:
+      - "!all"
+      - "!min"
+      - network
+  delegate_to: "{{ (_edpm_hosts | dict2items | first).key }}"
+  run_once: true
+
+- name: Update the hosts file on the Ceph bootstrap host
+  become: true
+  ansible.builtin.lineinfile:
+    path: /etc/hosts
+    line: "{{ ceph_boot_ssh_ip }} {{ (_edpm_hosts | dict2items | first).key }}"
+    state: present
+    create: true
+    backup: true
+    insertbefore: EOF
+  vars:
+    ceph_boot_ssh_ip: "{{ ansible_all_ipv4_addresses | ansible.utils.ipaddr(_subnet_network_range) | first }}"
+  delegate_to: "{{ (_edpm_hosts | dict2items | first).key }}"
+  run_once: true
+
+- name: Ensure Ceph bootstrap host can ping itself
+  ansible.builtin.command:
+    cmd: |
+      ping -c1 "{{ (_edpm_hosts | dict2items | first).key }}"
+  register: _cmd_result
+  retries: 5
+  delay: 60
+  until: _cmd_result.rc == 0
+  delegate_to: "{{ (_edpm_hosts | dict2items | first).key }}"
+
+- name: Create Ceph playbook variables file
+  ansible.builtin.copy:
+    dest: "~/ci-framework-data/parameters/ceph-{{ _az }}.yml"
+    content: |
+      ---
+      cifmw_cephadm_cluster: {{ _az }}
+      ssh_network_range: {{ _subnet_network_range }}
+      cifmw_ceph_target: {{ _group_name }}
+      storage_network_range: {{ _storage_network_range | ansible.utils.ipaddr('network/prefix') }}
+      storage_mgmt_network_range: {{ _storage_mgmt_network_range |  ansible.utils.ipaddr('network/prefix') }}
+      cifmw_ceph_client_service_values_post_ceph_path_dst: /tmp/edpm_service_values_post_ceph_{{ _az }}.yaml
+      cifmw_ceph_client_values_post_ceph_path_dst: "{{ _arch_deployment_path }}/values.yaml"
+      cifmw_ceph_spec_data_devices: >-
+        data_devices:
+          all: true
+      cifmw_ceph_client_vars: /tmp/ceph_client_{{_az}}.yml
```

Comment:
The mode has been set. 

cifmw_ceph_spec_data_devices is embedded yaml so it needs the block scalar so its format is unchanged.

https://github.com/openstack-k8s-operators/ci-framework/blob/9a70c3692f7d4dc9fe05da82d64e667cec92638d/roles/cifmw_ceph_spec/README.md?plain=1#L26-L28

It's a small amount of yaml (with embedded yaml). When I pass through  `| to_nice_yaml` the formatting for the embedded yaml looks like this.

```yaml
cifmw_ceph_client_service_values_post_ceph_path_dst: /tmp/edpm_service_values_post_ceph_az0.yaml
cifmw_ceph_client_values_post_ceph_path_dst: /tmp/values.yaml
cifmw_ceph_client_vars: /tmp/ceph_client_az0.yml
cifmw_ceph_spec_data_devices: "data_devices:\n  all: true"
cifmw_ceph_target: bar
cifmw_cephadm_cluster: az0
ssh_network_range: 172.18.0.0/24
storage_mgmt_network_range: 172.20.0.0/24
storage_network_range: 172.18.0.0/24
```
We'll know if it proper yaml because the ceph playbook uses it as a vars file.

Reply:
The latest update contains the change you requested with `| to_nice_yaml`

https://github.com/openstack-k8s-operators/ci-framework/compare/7e2d8f880b0d9459cc30a2a5bbc4fe63dd613eb9..582f14e3ae5be161d9830d81a536a97cd1671213